lib/Base/Image.cpp

```diff
@@ -459,18 +483,42 @@ void glow::loadImagesAndPreprocess(const llvm::ArrayRef<std::string> &filenames,
   case ImageLayout::NHWC:
     batchDims = {numImages, imgHeight, imgWidth, numChannels};
     break;
+  default:
+    LOG(FATAL) << "Unexpected layout\n";
   }
-  inputImageData->reset(ElemKind::FloatTy, batchDims);
-  auto IIDH = inputImageData->getHandle<>();
+  inputImageData.reset(ElemKind::FloatTy, batchDims);
+  auto IIDH = inputImageData.getHandle<>();
 
   // Read images into local tensors and add to batch.
   for (size_t n = 0; n < filenames.size(); n++) {
     Tensor localCopy =
         readPngImageAndPreprocess(filenames[n], imageNormMode,
                                   imageChannelOrder, imageLayout, mean, stddev);
     DCHECK(std::equal(localCopy.dims().begin(), localCopy.dims().end(),
-                      inputImageData->dims().begin() + 1))
+                      inputImageData.dims().begin() + 1))
         << "All images must have the same dimensions";
     IIDH.insertSlice(localCopy, n);
   }
 }
+
+/// Dispatching loading to the format handlers.
+void glow::loadImagesAndPreprocess(const llvm::ArrayRef<std::string> &filenames,
+                                   Tensor *inputImageData,
+                                   ImageNormalizationMode imageNormMode,
+                                   ImageChannelOrder imageChannelOrder,
+                                   ImageLayout imageLayout) {
+  DCHECK(!filenames.empty())
+      << "There must be at least one filename in filenames.";
+  assert((dim_t)filenames.size() == filenames.size());
```

Comment:
nit: A little funny here mixing `DCHECK` and `assert`, I'd probably just move this one to `DCHECK_EQ`